powerhome/playbook
# [PLAY-1082]  Linter implementation: kits A - D

[PLAY-1082](https://nitro.powerhrg.com/runway/backlog_items/PLAY-1082)
[Nitro PR
](https://github.com/powerhome/nitro-web/pull/36411)[Alpha Nitro Docs](https://pr36411.nitro-web.beta.px.powerapp.cloud/dev_docs/playbook)

This PR includes the linter fixes for the kits from A to D, except for the Date Picker Kit.

- I used the command `yarn eslint './playbook/app/pb_kits/playbook/pb_c*/**/*.{js,jsx,ts,tsx}’` to output the errors and warnings of the kits. All the errors were fixed and most of the warnings too.
- Kits that didn’t have a return type were mostly typed with `React.ReactElement`
- The `Function` type in the `htmlOptions` prop was replaced by `() => void`
- Properties typed with `Object` were mostly replaced by `{[key: string]: string}`
- The biggest changes were in the [pbChartsDarkTheme](https://github.com/powerhome/playbook/pull/2942/files#diff-3ce93503d79ac334df144584e708d6553998beea173fc6a4f1fb6cc52a078f6e) and [pbChartsLightTheme](https://github.com/powerhome/playbook/pull/2942/files#diff-e097999d3cdbb8eec0c4f93f624d7225d4e0fe7722abcc576cee3a0bbc353c11), where some properties were removed because they were not available in the typing.
- A declaration file for the tokens was created. [Check it here](https://github.com/powerhome/playbook/pull/2942/files#diff-a7e8e26204b81576594e0f6bbdc8c1222f13b3e994273884bc7dc401963f1eab).
- The date picker kit is not included in this PR. We need to work on a separate PR, after linting its files, the tests started to fail.

#### How to test:
- Run the following command for each of kit group from a to d - replace `pb_c` with the other initials: `yarn eslint './playbook/app/pb_kits/playbook/pb_c*/**/*.{js,jsx,ts,tsx}’` 
- There should not be any error in the output
- Run the command `./test.sh` inside the playbook folder. All the tests must pass.
- Go to the [`dev_docs/playbook` ](https://pr36411.nitro-web.beta.px.powerapp.cloud/dev_docs/playbook)route in nitro's milano build and test the kits 

#### Checklist:
